lib/webpacker/dev_server.rb

```diff
@@ -6,10 +6,14 @@ def initialize(webpacker)
   end
 
   def running?
-    Socket.tcp(host, port, connect_timeout: 1).close
-    true
-  rescue Errno::ECONNREFUSED, NoMethodError
-    false
+    return @running if defined?(@running)
+
+    @running = begin
+        Socket.tcp(host, port, connect_timeout: 0.1).close
```

Comment:
In my testing (https://github.com/rails/webpacker/issues/745#issuecomment-326973005), `connect_timeout` could be even shorter. Could start with `0.01` and make it configurable (with a `mattr_accessor`) to accommodate slower systems. Then memoization won't be necessary.

Reply:
Yeah, I was bit afraid as well but wanted to get your take on memoization. Pushed an update with configurable timeout 👍